### PR TITLE
Add chrome.storage.sync.config

### PR DESCRIPTION
**Pull request by michielbdejong**
TO DO:
- [x] Add test for call to `setInterval`
- [x] Input checking
- [x] Remove `console.log` statement
- [x] ~~Change `console.error` call to proper error management (see https://github.com/Kinto/kinto-chrome/issues/2)~~ -> follow-up
- [x] Remove use of `btoa` for the default config
- [x] Add docs
- [x] Rebase the two unrelated commits out of this branch

### CODE REVIEW

**Comment by michielbdejong** [comment]
Actually syncing the data is blocked by https://github.com/Kinto/kinto.js/issues/306 so I'll reduce this PR in scope to just setting the config and creating the timer (not actually successfully syncing anything)

**Comment by michielbdejong** [comment]
Will create follow-up tickets for:
- make sync work and add unit tests for it
- remove the IdSchema to allow any string as item key

**Comment by michielbdejong** [comment]
Rebased on #8 (will request review of this PR after that one is merged).

File: lib/chrome.js [hunk 1]
```diff
@@ -86,6 +80,41 @@ export class StorageArea {
...
+
+    this._config = config;
+
+    if (typeof config.interval === "number") {
+      this._sync().then(() => {
+        this.syncTimer = setInterval(() => {
+          this._sync().catch(err => {
+            console.error("Sync Error", err.message);
+          });
+        }, config.interval);
```
Comment (n1k0) [inline lib/chrome.js:117]: Don't we want a check for a minimum amount of time here?

File: test/chrome_test.js [hunk 2]
```diff
@@ -47,9 +48,25 @@ describe("StorageArea", function() {
...
+    it("should set the config", function() {
+      const area = new StorageArea("sync");
+      area.config = { type: "kinto", remote: "http://localhost:8080/v1/" };
+      expect(area.config.remote).to.eql("http://localhost:8080/v1/");
+    });
+
+    it("should configure a sync timer", function(done) {
+      const area = new StorageArea("sync");
+      area.config = { type: "kinto", interval: 123000 };
+      setTimeout(function() {
```
Comment (n1k0) [inline test/chrome_test.js]: You should use sinon fake timers and sandbox.clock.tick() here http://sinonjs.org/docs/#clock

**Comment by n1k0** [comment]
Looks good, r=r+wc

File: lib/chrome.js (hunk 1 above)
Comment (michielbdejong) [inline lib/chrome.js:117]: ok, added a check for >= 1000 ms

File: test/chrome_test.js (hunk 2 above)
Comment (michielbdejong) [inline test/chrome_test.js]: This is not really a timer, it's just waiting for https://github.com/michielbdejong/kinto-chrome/blob/chrome-storage-sync-config/lib/chrome.js#L89-L92 to resolve its promise. I changed it to `Promise.resolve().then` to make this clearer, and added a TODO comment which I plan to take care of in #12.

**Pull request merged**